nixos/modules/services/misc/matrix-registration.nix

```diff
@@ -0,0 +1,218 @@
+{ config, pkgs, lib, ... }:
+
+with lib;
+let
+  dataDir = "/var/lib/matrix-registration";
+  cfg = config.services.matrix-registration;
+  format = pkgs.formats.yaml { };
+  matrix-registration-config =
+    format.generate "config.yaml" cfg.settings;
+  matrix-registration-cli-wrapper = pkgs.stdenv.mkDerivation {
+    name = "matrix-registration-cli-wrapper";
+    buildInputs = [ pkgs.makeWrapper ];
+    buildCommand = ''
+      mkdir -p $out/bin
+      makeWrapper ${pkgs.matrix-registration}/bin/matrix-registration "$out/bin/matrix-registration" \
+        --add-flags "--config-path='${matrix-registration-config}'"
+    '';
+  };
+
+in
+{
+  options.services.matrix-registration = {
+    enable = mkEnableOption "Start Matrix Registration";
```

Comment:
`mkEnableOption` description is `description = "Whether to enable ${na
me}.";`.

Reply:
Should I end every description with '.'? I've seen both so I am unsure.

Reply:
- `mkEnableOption` - do not end with a period, it is included already.
- `mkOption` - always end with a period, it is not included for you.

Reply:
huh okay. then I'll add the missing dots to all the `mkOption`s later